qa/integration-tests/src/test/java/io/camunda/zeebe/it/management/RebalancingEndpointIT.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.it.management;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.camunda.zeebe.client.ZeebeClient;
+import io.camunda.zeebe.client.api.response.PartitionInfo;
+import io.camunda.zeebe.test.util.testcontainers.ZeebeTestContainerDefaults;
+import io.zeebe.containers.cluster.ZeebeCluster;
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpRequest.BodyPublishers;
+import java.net.http.HttpResponse.BodyHandlers;
+import java.time.Duration;
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.testcontainers.containers.Network;
+
+final class RebalancingEndpointIT {
+  private final Network network = Network.newNetwork();
+
+  private final ZeebeCluster cluster =
+      ZeebeCluster.builder()
+          .withImage(ZeebeTestContainerDefaults.defaultTestImage())
+          .withEmbeddedGateway(true)
+          .withBrokersCount(3)
+          .withPartitionsCount(3)
+          .withReplicationFactor(3)
+          .withNetwork(network)
+          .build();
+
+  private ZeebeClient client;
+
+  @BeforeEach
+  void setup() {
+    cluster.start();
+    client = cluster.newClientBuilder().build();
+  }
+
+  @AfterEach
+  void teardown() {
+    cluster.stop();
+  }
+
+  @Test
+  void shouldRebalanceCluster() throws IOException, InterruptedException {
+    // given
+    badLeaderDistribution();
+
+    // when
+    triggerRebalancing();
+
+    // then
+    reachesGoodLeaderDistribution();
+  }
+
+  private void triggerRebalancing() throws IOException, InterruptedException {
+    final var gateway = cluster.getGateways().values().stream().findFirst().orElseThrow();
+    final var monitoringAddress = gateway.getExternalMonitoringAddress();
+    final var httpClient = HttpClient.newHttpClient();
+    final var request =
+        HttpRequest.newBuilder()
+            .POST(BodyPublishers.noBody())
+            .uri(URI.create("http://" + monitoringAddress + "/actuator/rebalance"))
+            .build();
+    final var response = httpClient.send(request, BodyHandlers.discarding());
+    assertThat(response.statusCode()).isEqualTo(200);
+  }
+
+  private void badLeaderDistribution() {
+    cluster.getBrokers().get(1).stop();
```

Comment:
That's fair, when I wrote this I thought it would be fine because if that's not the case then we have already achieved our goal: a bad leader distribution. But if broker 1 here is leader of two partitions and we restart it, we might reach a good distribution and the test fails.

I'll change it then to only restart a broker if we have a good distribution and do nothing otherwise. Good catch, I bet this would have showed up in a few weeks as a flaky test.